### PR TITLE
cdrom_media: Fix device tray not open issue

### PR DESCRIPTION
**Pull request by leidwang**
Fix device tray not open issue

Signed-off-by: Leidong Wang <leidwang@redhat.com>

ID:1898030,1898032

### CODE REVIEW

**Comment by leidwang** [comment]
JOB ID     : 04436de93744eef4156fdf1dc10424c0eefda863
JOB LOG    : /root/avocado/job-results/job-2020-12-21T00.55-04436de/job.log
 (1/2) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.eject_media.force_eject.q35: PASS (60.79 s)
 (2/2) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.change_media.q35: PASS (53.30 s)

**Comment by leidwang** [comment]
JOB ID     : 2224a1d35bf8b2212699cf09ad5ff23c99616b82
JOB LOG    : /root/avocado/job-results/job-2020-12-21T01.27-2224a1d/job.log
 (1/2) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.eject_media.force_eject.q35: PASS (64.49 s)
 (2/2) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.change_media.q35: PASS (52.32 s)

**Comment by leidwang** [comment]
@qingwangrh Could you please review this patch?Thanks a lot!

File: qemu/tests/change_media.py [hunk 1]
```diff
@@ -89,9 +91,6 @@ def get_qdev_by_filename(filename):
         msg = "Fail to insert device %s to guest" % orig_img_name
         test.fail(msg)
 
-    if check_block_locked(device_name):
-        test.fail("Unused device is locked.")
-
```
Comment (qingwangrh) [inline qemu/tests/change_media.py:94]: why remove this?

Comment (leidwang) [reply qemu/tests/change_media.py:94]: As we discussed last week, different qemu versions have different `locked` states, so I think we can remove this checkpoint.

**Comment by qingwangrh** [comment]
python ConfigTest.py --testcase=eject_media,change_media --iothread_scheme=roundrobin --nr_iothreads=2 --platform=x86_64 --guestname=RHEL.8.4.0 --driveformat=virtio_blk,virtio_scsi --nicmodel=virtio_net --imageformat=qcow2 --machines=i440fx,q35 --clone=no

(01/12) Host_RHEL.m8.u4.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.eject_media.force_eject: PASS (54.14 s)
 (02/12) Host_RHEL.m8.u4.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.eject_media.force_eject.q35: PASS (50.28 s)
 (03/12) Host_RHEL.m8.u4.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.eject_media.eject: PASS (43.14 s)
 (04/12) Host_RHEL.m8.u4.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.eject_media.eject.q35: PASS (44.53 s)
 (05/12) Host_RHEL.m8.u4.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.change_media: PASS (32.25 s)
 (06/12) Host_RHEL.m8.u4.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.change_media.q35: PASS (38.98 s)
 (07/12) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.eject_media.force_eject: PASS (54.34 s)
 (08/12) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.eject_media.force_eject.q35: PASS (43.10 s)
 (09/12) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.eject_media.eject: PASS (42.90 s)
 (10/12) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.eject_media.eject.q35: PASS (43.73 s)
 (11/12) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.change_media: PASS (37.70 s)
 (12/12) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.change_media.q35: PASS (32.32 s)

LGTM
@YongxueHong  could you please help to review

**Comment by YongxueHong** [comment]
Please delete the last period in your commit header, then please add the reason about why fix issue by this way in your commit message.
Thanks.

File: qemu/tests/eject_media.py [hunk 2]
```diff
@@ -60,7 +60,7 @@ def eject_non_cdrom(device_name, force=False):
     # eject first time
     error_context.context("Eject original device.", logging.info)
     with eject_check:
-        vm.eject_cdrom(device_name)
+        vm.eject_cdrom(device_name, force=True)
```
[the remark this answers is not in the document]
Comment (qingwangrh) [reply qemu/tests/eject_media.py:63]: This issue duo to qemu or guest behavior changed, i suggest it need to check the lock status in advance.
 if locked then set force to True. same behavior may apply to blockdev_change_media. detail may refer to  https://bugzilla.redhat.com/show_bug.cgi?id=1820506

[the remark this answers is not in the document]
Comment (YongxueHong) [reply qemu/tests/eject_media.py:63]: Thanks @qingwangrh 
I didn't any info about this from the commit message, so it make me confused.
Please summary the above information in your commit message @leidwang 
thanks,

**Comment by leidwang** [comment]
> Please delete the last period in your commit header, then please add the reason about why fix issue by this way in your commit message.
> Thanks.
> 
> You could refer to [avocado-framework/avocado-vt@100c5f4](https://github.com/avocado-framework/avocado-vt/commit/100c5f43c988cf644494a7736cffc8b8d400153f)

Thank you for your guidance and updated the commit message.Could you please review this patch again?

**Review by YongxueHong — APPROVED** [review]

**Pull request merged**